kms/keys.go

```diff
@@ -0,0 +1,234 @@
+package kms
+
+import (
+	"go.uber.org/zap"
+	"google.golang.org/api/cloudkms/v1"
+)
+
+type CryptoKey struct {
+	ID               uint `gorm:"primarykey"`
+	ProjectID        string
+	CreateTime       string
+	Name             string
+	NextRotationTime string
+
+	PrimaryAlgorithm string
+
+	CaviumCerts          []*KeyCertChainsCaviumCerts           `gorm:"constraint:OnDelete:CASCADE;"`
+	GoogleCardCerts      []*CryptoKeyCertChainsGoogleCardCerts `gorm:"constraint:OnDelete:CASCADE;"`
+	GooglePartitionCerts []*KeyCertChainsGooglePartitionCerts  `gorm:"constraint:OnDelete:CASCADE;"`
+
+	PrimaryAttestationContent string
+	PrimaryAttestationFormat  string
+
+	PrimaryCreateTime       string
+	PrimaryDestroyEventTime string
+	PrimaryDestroyTime      string
+
+	PrimaryExternalProtectionLevelOptionsExternalKeyUri string
+
+	PrimaryGenerateTime        string
+	PrimaryImportFailureReason string
+	PrimaryImportJob           string
+	PrimaryImportTime          string
+	PrimaryName                string
+	PrimaryProtectionLevel     string
+	PrimaryState               string
+
+	Purpose        string
+	RotationPeriod string
+
+	VersionTemplateAlgorithm       string
+	VersionTemplateProtectionLevel string
+}
+
+func (CryptoKey) TableName() string {
+	return "gcp_kms_crypto_keys"
+}
+
+type KeyCertChainsCaviumCerts struct {
+	CryptoKeyID                  uint `gorm:"primarykey"`
+	CryptoKeyCertificateChainsID uint
```

Comment:
Those two lines should be:

`ID uint gorm:"primarykey"`
`CryptoKeyID uint` (this is the foreign key)